pkg/apis/topology/v1alpha1/types_partitionset.go

```diff
@@ -58,16 +60,47 @@ type PartitionSetSpec struct {
 
 	// +optional
 
-	// selector (optional) is a label selector that filters shard targets.
-	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+	// MatchExpressions (optional) specifies filtering for shard targets.
+	// MatchExpressions are used here instead of Selector to avoid possible collisions with
+	// the MatchLabels generated for the Partition according to the dimensions in the PartitionSet.
+	// This does not limit the filtering as any MatchLabel can also be expressed as
+	// a MatchExpression.
+	MatchExpressions []metav1.LabelSelectorRequirement `json:"matchExpressions,omitempty"`
```

Comment:
Also could we rename it to `shardSelector/shardMatchExpression` ?

Reply:
as per the comment " MatchExpressions are used here instead of Selector to avoid possible collisions with the MatchLabels generated for the Partition".
PartitionSets are used to generate Partitions. Partitions have a LabelSelector. LabelSelectors have two parts:
-  MatchLabels
- MatchExpressions

MatchLabels of Partitions are generated from the Dimension field of PartitionSet. MatchExpressions of Partitions are copied from MatchExpressions of PartitionSet. Keeping  this separation provides a clearer UX. It would be more difficult for users to understand the generated Partitions from a PartitionSet if MatchLabels could be a mix of  the Dimension generation and MatchLabels inherited from the PartitionSet. As per the comment there is no trade-off in terms of capabilities: any MatchLabels can be expressed by MatchExpressions.

Reply:
> as per the comment " MatchExpressions are used here instead of Selector to avoid possible collisions with the MatchLabels generated for the Partition".

if we have a collision, does it mean we can achieve the same by providing a selector on a `Partition` resource :) ?

Reply:
If I understand your point correctly PartitionSet is just a convenient way to create Partitions. Service Providers are free to create Partitions the way it best suits them: using PartitionSet or directly creating the resource. If they do it manually they can set Selectors: MatchLabels and MatchExpressions in Partitions in a similar way to what is generated by the controller.